app/models/user/reputation_tokens/code_contribution_token.rb

```diff
@@ -1,11 +1,13 @@
 class User::ReputationTokens::CodeContributionToken < User::ReputationToken
-  params :repo, :pr_node_id, :pr_number, :pr_title
+  params :repo, :pr_node_id, :pr_number, :pr_title, :merged_at
   category :building
   reason :contributed_code
   levels %i[minor regular major]
   values({ minor: 5, regular: 12, major: 30 })
 
   before_validation on: :create do
+    self.earned_on = self.merged_at unless earned_on
```

Comment:
The `unless earned_on` will stop us from updating these records. Probably not an issue moving forward but something to be aware of if you do this then rerun the script to try and get the right data in. Probably best left as you have it but I just wanted to raise it.

Reply:
Yeah I considered it to. The reasoning was that `merged_at` shouldn't ever change.